build/paramfetch.go

```diff
@@ -46,35 +59,61 @@ func GetParams(storage bool) error {
 			continue
 		}
 
-		if err := maybeFetch(name, info); err != nil {
-			return err
-		}
+		ft.maybeFetchAsync(name, info)
 	}
 
-	return nil
+	return ft.wait()
 }
 
-func maybeFetch(name string, info paramFile) error {
-	path := filepath.Join(paramdir, name)
-	f, err := os.Open(path)
-	if err == nil {
-		defer f.Close()
+func (ft *fetch) maybeFetchAsync(name string, info paramFile) {
+	ft.wg.Add(1)
 
-		h := blake2b.New512()
-		if _, err := io.Copy(h, f); err != nil {
-			return err
+	go func() {
+		defer ft.wg.Done()
+
+		path := filepath.Join(paramdir, name)
+
+		err := ft.checkFile(path, info)
+		if !os.IsNotExist(err) && err != nil {
+			log.Warn(err)
+		}
+		if err == nil {
+			return
 		}
 
-		sum := h.Sum(nil)
-		strSum := hex.EncodeToString(sum[:16])
-		if strSum == info.Digest {
-			return nil
+		ft.fetchLk.Lock()
+		defer ft.fetchLk.Unlock()
+
+		if err := doFetch(path, info); err != nil {
+			ft.errs = append(ft.errs, xerrors.Errorf("fetching file %s: %w", path, err))
```

Comment:
This means we are fetching only one file at the time. Do we want to fetch multiple?

Reply:
Only one for now (main reason being the progress bar being too anoying to fix)